src/sql/parts/notebook/models/cell.ts

```diff
@@ -0,0 +1,321 @@
+
```

Comment:
This class is essentially being ported as-is with changes just to recompile against internal vs. extension APIs. May need refresh since we took some small changes in the extension.